src/pieces/rook.js

```diff
@@ -1,18 +1,39 @@
 import Piece from './piece';
-
 class Rook extends Piece {
   constructor(row, column, side) {
     super(row, column, side);
     this.name = 'rook';
     this.display = `<i class="fas fa-chess-rook ${side}"></i>`; //fontawesome rook
   }
-  findLegalMoves() {
-    const possibleMoves = [];
-  for (let i = 1; i < 8; i++) {
-      if (this.row - i >= 0) possibleMoves.push([this.row - i, this.column]);
-      if (this.row + i < 8) possibleMoves.push([this.row + i, this.column]);
-      if (this.column - i >= 0) possibleMoves.push([this.row, this.column - i]);
-      if (this.column + i < 8) possibleMoves.push([this.row, this.column + i]);
+  
+  findLegalMoves(board) {
```

Comment:
Jest git, normalnie bym się nie czepiał ale jako że obiecałem że na to spojrzę to wrzucam swoją wersje:
```suggestion
   findLegalMoves(board) {
    const possibleMoves = [];

    const directions = [
      { row: -1, column: 0 }, // Up
      { row: 1, column: 0 },  // Down
      { row: 0, column: -1 }, // Left
      { row: 0, column: 1 }   // Right
    ];

    for (const { row, column } of directions) {
      for (let distance = 1; distance < 8; distance++) {
        const newRow = this.row + distance * row;
        const newColumn = this.column + distance * column;
        const destination = [newRow, newColumn]

        const isOutOfBounds = destination.some(value => value < 0 || value >= 8)
        if (isOutOfBounds) break;

        const square = board.getSquare(newRow, newColumn);
        const squareIsEmpty = !square.piece
        const squareIsOccupiedByTheOpponentsPiece = square.piece?.side !== this.side

        if (squareIsEmpty) {
          possibleMoves.push(destination);
          continue;
        }

        if (squareIsOccupiedByTheOpponentsPiece) {
          possibleMoves.push(destination);
          break;
        }

        if (!squareIsOccupiedByTheOpponentsPiece) break;

      }
    }
    return possibleMoves;
  }
```

Reply:
Kluczowe zmiany:

- ułożyłem if'y inaczej żeby bardziej podstawowy przypadek był sprawdzony wcześniej (wykorzystalem keyword continue, żeby pominąć iteracje i przejść do następnej)
- przemianowałem nazwy zmiennych tak by unikać skrótów
-  wyciagnełem warunki do zmiennych by móc nadać im nazwe opisującą co dany warunek sprawdza